### PR TITLE
Add cache with multiple shards to decrease lock contention

### PR DESCRIPTION
**Pull request by fgrzadkowski**
This change improves latency of read operations.

### CODE REVIEW

File: pkg/util/cache.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package util
```
Comment (wojtek-t) [inline pkg/util/cache.go]: Can you please add some unit test?

**Comment by wojtek-t** [comment]
Generally looks good - please add a unit test for the cache.

File: pkg/util/cache.go (hunk 1 above)
Comment (fgrzadkowski) [inline pkg/util/cache.go]: Done.

**Comment by wojtek-t** [comment]
LGTM - will merge on green.

**Comment by nikhiljindal** [comment]
Travis is green. Merging.

**Pull request merged**